2024-10-29 13:30:41.846 - INFO    - MainThread - client.py:204 [connectAsync] - Connecting to 127.0.0.1:7497 with clientId 22559...
2024-10-29 13:30:41.848 - INFO    - MainThread - client.py:212 [connectAsync] - Connected
2024-10-29 13:30:41.856 - INFO    - MainThread - client.py:341 [_onSocketHasData] - Logged on to server version 176
2024-10-29 13:30:41.859 - INFO    - MainThread - wrapper.py:1111 [error] - Warning 2104, reqId -1: Market data farm connection is OK:cashfarm
2024-10-29 13:30:41.860 - INFO    - MainThread - wrapper.py:1111 [error] - Warning 2104, reqId -1: Market data farm connection is OK:usfuture
2024-10-29 13:30:41.860 - INFO    - MainThread - wrapper.py:1111 [error] - Warning 2104, reqId -1: Market data farm connection is OK:usfarm.nj
2024-10-29 13:30:41.864 - INFO    - MainThread - wrapper.py:1111 [error] - Warning 2104, reqId -1: Market data farm connection is OK:usfarm
2024-10-29 13:30:41.864 - INFO    - MainThread - wrapper.py:1111 [error] - Warning 2106, reqId -1: HMDS data farm connection is OK:ushmds
2024-10-29 13:30:41.864 - INFO    - MainThread - wrapper.py:1111 [error] - Warning 2158, reqId -1: Sec-def data farm connection is OK:secdefhk
2024-10-29 13:30:41.865 - INFO    - MainThread - client.py:218 [connectAsync] - API connection ready
2024-10-29 13:30:41.967 - ERROR   - MainThread - wrapper.py:1113 [error] - Error 321, reqId -1: Error validating request.-'cr' : cause - The API interface is currently in Read-Only mode.
2024-10-29 13:30:42.065 - ERROR   - MainThread - wrapper.py:1113 [error] - Error 321, reqId -1: Error validating request.-'b2' : cause - The API interface is currently in Read-Only mode.
2024-10-29 13:30:45.874 - ERROR   - MainThread - ib.py:1779 [connectAsync] - open orders request timed out
2024-10-29 13:30:45.874 - ERROR   - MainThread - ib.py:1779 [connectAsync] - completed orders request timed out
2024-10-29 13:30:45.876 - INFO    - MainThread - ib.py:1789 [connectAsync] - Synchronization complete
